src/ralph/networks/filters.py

```diff
@@ -0,0 +1,37 @@
+import ipaddress
+
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.admin.filters import TextListFilter
+
+
+class IPRangeFilter(TextListFilter):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.title = _('IP range')
+
+    def queryset(self, request, queryset):
+        if self.value():
+            network = ipaddress.ip_network(self.value())
+            min_ip = int(network.network_address)
+            max_ip = int(network.broadcast_address)
+
+            queryset = queryset.filter(number__gte=min_ip, number__lte=max_ip)
+        return queryset
+
+
+class NetworkRangeFilter(TextListFilter):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.title = _('Network range')
+
+    def queryset(self, request, queryset):
+        if self.value():
+            network = ipaddress.ip_network(self.value())
+            min_ip = int(network.network_address)
+            max_ip = int(network.broadcast_address)
+
+            queryset = queryset.filter(min_ip__gte=min_ip, max_ip__lte=max_ip)
```

Comment:
how will this work with this whole 'parent' filter? IMO when I filter by this, I want to see all networks which falls into this range, not only top-level ones. 

Maybe parent filter should be replaced by that + another (boolean) filter "show only top level networks"?